shape.c

```diff
@@ -733,9 +733,10 @@ rb_shape_get_next(rb_shape_t *shape, VALUE obj, ID id)
                 if (RCLASS_EXT(klass)->variation_count >= SHAPE_MAX_VARIATIONS) {
                     rb_category_warn(
                         RB_WARN_CATEGORY_PERFORMANCE,
-                        "Maximum shapes variations (%d) reached by %"PRIsVALUE", instance variables accesses will be slower.",
-                        SHAPE_MAX_VARIATIONS,
-                        rb_class_path(klass)
+                        "The class %"PRIsVALUE" reached %d shape variations, instance variables accesses will be slower and memory usage increased.\n"
+                        "It is recommended to define instance variable in a consistent order, for instance by eagerly defining them all in the `initialize` method.",
```

Comment:
Minor, but backquotes seem rare in Ruby exception messages, here I think `#initialize` would be better.

Reply:
https://github.com/ruby/ruby/pull/9280